src/main/java/bio/terra/cli/businessobject/User.java

```diff
@@ -154,11 +180,25 @@ public static void login() {
     }
   }
 
+  /**
+   * Fetches a Google Application Credentials with {@code PET_SA_SCOPES} and set to {@code
+   * googleCredentials}.
+   */
+  private void loadAppDefaultCredentials() {
+    if (googleCredentials == null) {
+      useApplicationDefaultCredentials = true;
```

Comment:
Can you login with ADC, and then login with the regular OAuth flow? I'm not sure, but it looks like you set this to true and there's no way to get it set back to false.

If that's the case, perhaps you can set this property in the `login` command class. e.g. If they specify the use ADC flag, then call `User.setUseADC(true)` before calling `login`. And if they don't specify that flag, then call `User.setUseADC(false)` before calling `login`.

Reply:
thanks. that is a very good point. setUseAdc needs to be a static method to be called in `login` command class. So instead I reuse my current structure and set `useAdc` field in `login`. When `login(false)`, useAdc field is set to fales and vice versa. 

i tried terra auth login, and then terra auth login --use-app-default-credential and then terra auth login, I'm able to switch between the two